variables.tf

```
@@ -73,6 +73,47 @@ variable "cloud_monitoring_instance_crn" {
   default     = null
 }
 
+##############################################################
+# CSPM
+##############################################################
+
+variable "cspm_enabled" {
+  description = "Enable Cloud Security Posture Management (CSPM) for the Workload Protection instance. This will create a trusted profile for the App Config instance and associate it with the Workload Protection instance."
+  type        = bool
+  default     = true
+  nullable    = false
+}
+
+variable "app_config_crn" {
+  description = "The CRN of the App Config instance to use with the Workload Protection instance. Can be `null` if `cspm_enabled` is not enabled. Must be a valid App Config CRN."
+  type        = string
+  default     = null
+  validation {
+    condition     = var.cspm_enabled ? var.app_config_crn != null : true
+    error_message = "Cannot be `null` if CSPM is enabled. Must be a valid App Config CRN."
+  }
+  validation {
+    condition = anytrue([
+      can(regex("^crn:(.*:){3}apprapp:(.*:){2}[0-9a-fA-F]{8}(?:-[0-9a-fA-F]{4}){3}-[0-9a-fA-F]{12}::$", var.app_config_crn)),
+      var.app_config_crn == null,
+    ])
+    error_message = "The provided CRN is not a valid App Config CRN."
+  }
+}
+
+variable "scc_workload_protection_trusted_profile_name" {
+  description = "The name for the trusted profile that is created by this module. Must begin with a letter."
```

```suggestion
  description = "The name to give the trusted profile that is created by this module if `cspm_enabled` is `true. Must begin with a letter."
```

Can we also add variable validation:
- cannot be null if `cspm_enabled` is true
- Must begin with a letter

---

Also add to DA